util/src/main/java/org/folio/metastorage/util/OaiParser.java

```diff
@@ -0,0 +1,104 @@
+package org.folio.metastorage.util;
+
+import java.io.InputStream;
+import java.util.LinkedList;
+import java.util.List;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamConstants;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.XMLStreamReader;
+
+public class OaiParser {
+
+  private XMLInputFactory factory;
+
+  private String resumptionToken;
+
+  private List<String> identifiers = new LinkedList<>();
+
+  private List<String> records = new LinkedList<>();
+
+  public OaiParser() {
+    factory = XMLInputFactory.newInstance();
+    factory.setProperty(XMLInputFactory.SUPPORT_DTD, false);
+  }
+
+  /**
+   * Get resumption token.
+   * @return token string or null if none
+   */
+  public String getResumptionToken() {
+    return resumptionToken;
+  }
+
+  /**
+   * Get record identifiers.
+   * @return list of identifiers
+   */
+  public List<String> getIdentifiers() {
+    return identifiers;
+  }
+
+  /**
+   * Get record metadata content.
+   * @return list of metadata with item being null for deleted metadata
+   */
+  public List<String> getMetadata() {
```

Comment:
I think getIdentifier and getMetadata should be converted into `List<Records> getRecords()` with:

class Record {
   String metadata;
   String identifier;
   boolean isDeleted;
   Date datestamp;
    ...
}

I think it's better than conveying a proper state of "deteled" through a null value;

Reply:
Code easier to read and write. With verb=ListIdentifiers metadata would be null always, so the extra attribute is needed.